lib/validator/v3/validator.go

```diff
@@ -66,6 +67,13 @@ var (
 	// Hostname  have to be valid ipv4, ipv6 or strings up to 64 characters.
 	prometheusHostRegexp = regexp.MustCompile(`^[a-zA-Z0-9:._+-]{1,64}$`)
 
+	// global() cannot be used with other selectors.
+	andOr               = `(&&|\|\|)`
+	globalSelectorRegex = regexp.MustCompile(fmt.Sprintf(`%v global\(\)|global\(\) %v`, andOr, andOr))
```

Comment:
Hmm, I see you already have a test for `"global()||all()"`, so perhaps ` ` in the RE already stands for any amount of whitespace?  (Otherwise, how is that test passing?)

Reply:
We parse the selector string first and use the String() from that result for the validation matching. The parsing will trim out all of the whitespace.